frontend/apps/remark42/app/common/fetcher.test.ts

```diff
@@ -125,6 +150,133 @@ describe('fetcher', () => {
 
       expect(window.fetch).toHaveBeenCalledWith(apiUrl, { method: 'get', headers });
     });
+
+    it('should store JWT token in a cookie when received in header', async () => {
+      // Mock the auth cookie helper
+      const setAuthCookieSpy = jest.spyOn(cookies, 'setAuthCookie').mockImplementation(() => {});
```

Comment:
if function called only once per test I'd recommend using methods with `Once` suffix, so no `restoreAllMocks` is needed.
If test fails then the mock cleaning is not called and all the following test cases may be affected.

tip: you can perform expectations on `cookies.setAuthCookie` because the function is wrapped into the spy and the variable is not needed for the expectations.
```suggestion
      const setAuthCookieSpy = jest.spyOn(cookies, 'setAuthCookie').mockReturnValueOnce(undefined);
```
```suggestion
      const setAuthCookieSpy = jest.spyOn(cookies, 'setAuthCookie').mockImplementationOnce(jest.fn());
```